src/main/java/seedu/address/logic/commands/DeleteMeetingCommand.java

```diff
@@ -39,28 +39,25 @@ public DeleteMeetingCommand(Index index) {
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        List<Client> lastShownList = model.getFilteredClientList();
+        List<Client> lastShownClientList = model.getFilteredClientList();
+        List<Meeting> lastShownMeetingList = model.getFilteredMeetingList();
 
-        if (index.getZeroBased() >= lastShownList.size()) {
+        if (index.getZeroBased() >= lastShownMeetingList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_MEETING_DISPLAYED_INDEX);
         }
 
-        Client clientToEdit = lastShownList.get(index.getZeroBased());
+        Meeting meeting = lastShownMeetingList.get(index.getZeroBased());
+
+        Client clientToEdit = meeting.getClient();
         Client editedClient = new Client(
                 clientToEdit.getName(), clientToEdit.getPhone(), clientToEdit.getEmail(),
                 clientToEdit.getAddress(), clientToEdit.getTags()
         );
-        Meeting meeting = clientToEdit.getMeeting();
-
-        if (meeting == null) {
-            throw new CommandException(String.format(MESSAGE_MEETING_NOT_FOUND, clientToEdit));
-        }
 
         model.deleteMeeting(meeting);
```

Comment:
Not sure if it is checked, but seems like the check for null meetings is removed. `NoConflictMeetingList#remove` actually performs a check through `requireNonNull`to ensure no null arguments.

Reply:
I think this one not a big issue for now with the current implementation. Since the meeting is deleted from the meeting list, it has to exist in our clients. But maybe further down the road, we might need to change this with implementing client as a arraylist.

Reply:
If null is passed to `model#deleteMeeting`, a NullPointerException may be thrown. Maybe can test if we try to deleteMeeting from a Client that doesn't have a meeting?

Reply:
If meeting is indeed a null, can throw a `CommandException` to feedback to user that there are no meetings associated with that client

Reply:
`meeting` is now taken from `lastShownMeetingList`, so unless the user enter an index that is out of range (which will be catch by the code), `meeting` will not be null. Hence, I don't think a NullPointerException will be thrown. 
Testing whether we can do `deleteMeeting `from a client that doesn't have meeting, I am not too sure how to do that since we are accessing `meeting` from `lastShownMeetingList`.

Reply:
I mean if there is really a need for a check, I think it will be to check if 
`meeting.equals(clientToEdit.getMeeting())`.
But I don't think it is necessary. What do you think?

Reply:
@rexong didn't see your response before the 2 above. But yes, you are right that the meeting is deleted through the meeting list and this shouldn't affect functionality since there is a check in parser, so this is okay. Though I suspect in the future we may need to delete meetings through clients as that may be more natural for a user to do so. But for now, this is okay